### PR TITLE
Provide optional proxy instance

### PR DESCRIPTION
**Pull request by Hogman-the-Intruder**
Implements Terraform to (optionally) build an EC2 instance that will install mitmproxy and configure it to point to the Gigaproxy infra. This will allow users to only have to install the mitmproxy certificate on their client devices (or disable TLS/SSL/Cert validation). Refer to updated README for more info (`Optional Proxy Instance` section)

### CODE REVIEW

**Comment by Hogman-the-Intruder** [comment]
Ayo why did my alt account get the credit for this commit? That's fricked up

File: terraform/forwarder-lambda.tf
Comment (Hogman-the-Intruder) [inline terraform/forwarder-lambda.tf:1]: `terraform fmt --recursive` related change

File: terraform/provider.tf
Comment (Hogman-the-Intruder) [inline terraform/provider.tf:1]: `terraform fmt --recursive` related change

**Review by puzzlepeaches — APPROVED** [review]
SEND IT

**Pull request merged**